Block non-image uploads

salsa-nathan opened the pull request:
- Added `IAuthFunctions` to `plugin.py`
- Added chained auth functions for `user_update`, `organization_update` and `group_update` to check `request.files` mimetype

MarkCalvert commented:
Hi @salsa-nathan,
Do we also need to do a auth check for `user_create`, `organisation_create` and `group_create`? Or can you only upload files on the updates in CKAN?

salsa-nathan commented:
> Hi @salsa-nathan,
> Do we also need to do a auth check for `user_create`, `organisation_create` and `group_create`? Or can you only upload files on the updates in CKAN?

Thanks @MarkCalvert - I've added those now and also refactored the code a bit to make it cleaner.
